### PR TITLE
Eliminate non-fastcomp tests

### PR DESCRIPTION
**Pull request by aidanhs**
Stage 2 of #3243.

The first and second commits are fairly mechanical, there shouldn't be any issues there - mainly just reindenting.
The third commit mainly consisted of adding appropriate TODO messages.
The final commit actually removed some test content so is probably worth a look.

### CODE REVIEW

**Comment by aidanhs** [comment]
Just noticed my dreadfully named branch.

File: tests/test_core.py [hunk 1]
```diff
@@ -964,15 +959,15 @@ def ensure_stack_restore_count(function_name, expected_count):
     ensure_stack_restore_count('function _alloca_gets_restored', 1)
     ensure_stack_restore_count('function _stack_usage', 1)
 
+  # TODO: test only worked in non-fastcomp
```
Comment (kripken) [inline tests/test_core.py]: not quite - it works with `named == 0`

File: tests/test_core.py [hunk 2]
```diff
@@ -3019,10 +3000,9 @@ def test_runtimelink(self):
     Settings.RUNTIME_LINKED_LIBS = ['liblib.so'];
     self.do_run(main, 'supp: 54,2\nmain: 56\nsupp see: 543\nmain see: 76\nok.')
 
+  # TODO: test only worked in non-fastcomp
```
Comment (kripken) [inline tests/test_core.py]: this isn't a test, it's a utility. it also shouldn't call `self.skip`.

**Comment by aidanhs** [comment]
Addressed both comments. I guess can_dlfcn was previously problematic...

**Comment by kripken** [comment]
`tests/test_other.py` was broken, fixing now. I forgot to ask you to run some tests on this ;)

**Pull request merged**